web_external/js/models/DatasetModel.js

```diff
@@ -66,313 +39,153 @@ minerva.models.DatasetModel = minerva.models.MinervaModel.extend({
-                        this.trigger('m:dataLoaded', this.get('_id'));
-                    }, this)
-                });
-            }
+    /*
+     * Async function that loads any data needed by this dataset to render in GeoJs,
+     * sets that data as an attribute on this dataset named 'geoData',
+     * emitting the 'm:geoDataLoaded' event and passing a reference to this dataset
+     * when data is loaded or if this dataset did not need to load any data to render in GeoJs.
+     */
+    loadGeoData: function () {
+        var mm = this._initGeoRender();
+        if (this.get('geoData') !== null || mm.geo_render === null || !mm.geo_render.file_id) {
+            this.trigger('m:geoDataLoaded', this);
         } else {
-            if (this.latLongMapper) {
-
-                var geoJsonData = {
-                    type: 'FeatureCollection',
-                    features: []
-                };
-                // look at this.csv.data, an array
-                // create a point feature for each row
-                // use the mapper to get lat and long
-                _.each(this.csv.data, function (row) {
-                    var point = {
-                        type: 'Feature',
-                        // TODO need to get other property column, just hardcoding elevation for now
-                        properties: { elevation: Number(0) },
-                        geometry: {
-                            type: 'Point',
-                            coordinates: [Number(row[this.latLongMapper.longitudeColumn]), Number(row[this.latLongMapper.latitudeColumn])]
-                        }
-                    };
-                    geoJsonData.features.push(point);
-                }, this);
-                this.fileData = JSON.stringify(geoJsonData);
-                this.geoFileReader = 'jsonReader';
-                this.trigger('m:geoJsonDataLoaded', this.get('_id'));
-                this.trigger('m:dataLoaded', this.get('_id'));
-            }
-        }
-    },
-
-    geocodeTweet: function () {
-        girder.restRequest({
-            path: 'item/' + this.get('_id') + '/geocodetweet',
-            type: 'POST'
-        }).done(_.bind(function (resp) {
-            if (resp._id !== '') {
-                this.geojsonFileId = resp._id;
-                this.trigger('m:tweetGeocoded', this);
-            } else {
-                girder.events.trigger('g:alert', {
-                    icon: 'cancel',
-                    text: 'Could not geocode tweets.',
-                    type: 'error',
-                    timeout: 4000
-                });
-            }
-        }, this)).error(_.bind(function (err) {
-            console.error(err);
-            girder.events.trigger('g:alert', {
-                icon: 'cancel',
-                text: 'Could not geocode tweets.',
-                type: 'error',
-                timeout: 4000
+            var url = this._getGeoRenderDownloadUrl();
+            $.ajax({
```

Comment:
Why `$.ajax` and not `girder.restRequest`? I'm not fully aware of the context in which this is running, but I think this side steps Girder authentication?